source/sdk/threading/Thread.ooc

```diff
@@ -3,111 +3,49 @@ import native/[MutexUnix, MutexWin32]
 import native/[ThreadLocalUnix, ThreadLocalWin32]
 import native/[ConditionUnix, ConditionWin32]
 
-/**
- * A thread is a thread of execution in a program. Multiple threads
- * can run concurrently, allowing parallel computations to occur.
- *
- * However, since such threads are preemptible, synchronization issues
- * can occur, such as race conditions (two threads access the same memory
- * location in an interleaved manner, leaving an inconsistent state)
- * or
- */
 Thread: abstract class {
+	_code: Func
 
-    _code: Func
-
-    /**
-     * Create a new thread that will run a given function.
-     * Note that this only creates the thread - the start() method
-     * can be used to effectively start execution of this thread.
-     *
-     * @param code A function to be executed from the newly created
-     * thread. It can be a closure (and it's actually pretty convenient)
-     */
-    new: static func (._code) -> This {
-
-        version (unix || apple) {
-            return ThreadUnix new(_code) as This
-        }
-        version (windows) {
-            return ThreadWin32 new(_code) as This
-        }
-
-        Exception new(This, "Unsupported platform!\n") throw()
-        null
-
-    }
-
-    /**
-     * Starts the actual execution of a thread.
-     *
-     * This call is non-blocking, since the execution of this thread's
-     * code will happen concurrently.
-     *
-     * You can call wait() to block until the started thread has finished
-     * its job.
-     *
-     * @return true if the thread has started successfully, false otherwise
-     */
-    start: abstract func -> Bool
-
-    /**
-     * Blocks until this thread has finished running (ie. join this thread).
-     *
-     * @return true if the thread has successfully joined, ie. if it's finished running.
-     * otherwise, false - the thread is not joinable, is already dead, a deadlock was
-     * detected, etc.
-     */
-    wait: abstract func -> Bool
+	new: static func (._code) -> This {
+		version (unix || apple) {
+			return ThreadUnix new(_code) as This
```

Comment:
If `this` was enforced when referring to member variables, it would be more obvious that this refers to the incoming argument to the function, and not the member variable.